lib/plugins/aws/package/compile/events/msk/index.test.js

```diff
@@ -0,0 +1,242 @@
+    let eventSourceMappingResource;
+    let defaultIamRole;
+    let naming;
+
+    before(() =>
+      fixtures
+        .extend('function', {
+          functions: {
+            foo: {
+              events: [
+                {
+                  msk: {
+                    topic,
+                    arn,
+                  },
+                },
+              ],
+            },
+          },
+        })
+        .then(fixturePath =>
+          runServerless({
+            cwd: fixturePath,
+            cliArgs: ['package'],
+          }).then(({ awsNaming, cfTemplate }) => {
+            naming = awsNaming;
+            eventSourceMappingResource =
+              cfTemplate.Resources[
+                naming.getMSKEventLogicalId('foo', 'ClusterName', 'TestingTopic')
+              ];
+            defaultIamRole = cfTemplate.Resources.IamRoleLambdaExecution;
+          })
+        )
+    );
+
+    it('should correctly compile EventSourceMapping resource properties', () => {
+      expect(eventSourceMappingResource.Properties).to.deep.equal({
+        BatchSize: 100,
+        Enabled: true,
+        EventSourceArn: arn,
+        StartingPosition: 'TRIM_HORIZON',
+        Topics: [topic],
+        FunctionName: {
+          'Fn::GetAtt': [naming.getLambdaLogicalId('foo'), 'Arn'],
+        },
+      });
+    });
+
+    it('should update default IAM role with MSK statement', () => {
+      expect(defaultIamRole.Properties.Policies[0].PolicyDocument.Statement).to.deep.include({
+        Effect: 'Allow',
+        Action: ['kafka:DescribeCluster', 'kafka:GetBootstrapBrokers'],
+        Resource: [arn],
+      });
+    });
+
+    it('should update default IAM role with EC2 statement', () => {
+      expect(defaultIamRole.Properties.Policies[0].PolicyDocument.Statement).to.deep.include({
+        Effect: 'Allow',
+        Action: [
+          'ec2:CreateNetworkInterface',
+          'ec2:DescribeNetworkInterfaces',
+          'ec2:DescribeVpcs',
+          'ec2:DeleteNetworkInterface',
+          'ec2:DescribeSubnets',
+          'ec2:DescribeSecurityGroups',
+        ],
+        Resource: '*',
+      });
+    });
+
+    it('should correctly compile EventSourceMapping resource DependsOn ', () => {
+      expect(eventSourceMappingResource.DependsOn).to.equal('IamRoleLambdaExecution');
+    });
+  });
+
+  describe('when using all parameters', () => {
+    it('should correctly compile EventSourceMapping resource', () => {
+      const enabled = false;
+      const startingPosition = 'LATEST';
+      const batchSize = 5000;
+
+      return fixtures
+        .extend('function', {
+          functions: {
+            foo: {
+              events: [
+                {
+                  msk: {
+                    topic,
+                    arn,
+                    batchSize,
+                    enabled,
+                    startingPosition,
+                  },
+                },
+              ],
+            },
+          },
+        })
```

Comment:
For tests performance it'll be nice to rely on just one _fixture_, so use multiple events that resemble different tested configurations in scope of one service.